src/test/index.ts

```diff
@@ -309,10 +392,23 @@ export const test = _baseTest.extend<TestFixtures, WorkerAndFileFixtures>({
     }));
   }, { auto: true }],
 
-  context: async ({ browser, video, _artifactsDir }, use, testInfo) => {
+  _reuseBrowerContext: [new ReuseBrowerContextStorage(), { scope: 'worker' }],
+
+  context: async ({ browser, video, _artifactsDir, _reuseBrowerContext, _combinedContextOptions }, use, testInfo) => {
     const hook = hookType(testInfo);
     if (hook)
       throw new Error(`"context" and "page" fixtures are not supported in ${hook}. Use browser.newContext() instead.`);
+    if (_reuseBrowerContext.isEnabled()) {
+      let context: BrowserContext;
```

Comment:
Let your utility handle the hasContext branching. This removes `hasContext` & `setContext`.

`const context = await _reuseBrowerContext.obtainContext(browser, _combinedContextOptions);`